### ISSUE TITLE
`Format.pp_set_max_indent` with "too large" argument

### ISSUE THREAD

**Issue by sim642** [ocaml/ocaml#11517]
While trying to maximize the margin and max indent of `Format` to practically disable hard wrapping (but still make use of boxes, etc), I stumbled upon the following problem where max indent surprisingly did not get changed.

### Interactive session
For example, this is demonstrated by the following session:
```ocaml
OCaml version 4.14.0
Enter #help;; for help.

# Format.get_margin ();;
- : int = 78
# Format.get_max_indent ();;
- : int = 68
# Format.set_geometry ~margin:max_int ~max_indent:(max_int - 1);;
- : unit = ()
# Format.get_margin ();;
- : int = 1000000009
# Format.get_max_indent ();;
- : int = 68
```
Even though `set_geometry` is used, where the invariant validity check for `max_indent < margin` passes, as if it should work, `max_indent` isn't actually changed.

### Implementation
The issue is in `pp_set_max_indent`, which `set_geometry` uses:
https://github.com/ocaml/ocaml/blob/67bde9dba3cbfb1c1a17e6bb7d39def23e7a9460/stdlib/format.ml#L790-L792
Namely, after having truncated and set `state.pp_margin` to 1000000009, the argument `state.pp_margin - n` will be negative (`n` hasn't been truncated), so `pp_set_min_space_left` does nothing:
https://github.com/ocaml/ocaml/blob/67bde9dba3cbfb1c1a17e6bb7d39def23e7a9460/stdlib/format.ml#L779-L784

### Documentation
The documentation states the following about `pp_set_max_indent ppf d`:
https://github.com/ocaml/ocaml/blob/67bde9dba3cbfb1c1a17e6bb7d39def23e7a9460/stdlib/format.mli#L473-L477
In this case the latter applies (at least between setting the margin and setting the max indent), but actually both conditions are true (`max_int - 1` is also "too large") and contradict each other since the former implies that an _admissible_ value is picked instead. Technically such value exists: there are a few integers greater than 10^9 and less than the current margin 1000000009, hence _admissible_.

Given the currently implemented behavior, the "too large" condition on max indent is actually moot altogether because if the argument is greater than 1000000009 ("too large"), then it's also greater than any possible margin (which has the same notion of "too large") anyway.

This is related to #1525 but in the other extreme.

**Comment by xavierleroy** [comment]
Feel free to suggest a fix as a pull request, either changing the implementation to do something more predictable, or the documentation to warn against the problem.  I'm afraid the core dev team has more urgent things to work on right now.

**Comment by Octachron** [comment]
I think that updating the documentation is probably a good first step.

**Comment by olybeke** [comment]
hello @Octachron 
I'm Oluchi Nwankwo, an outreachy applicant and I would like to work on this issue. I have already set up my Ocaml environment. Can you assign me to work on this issue?

**Comment by Octachron** [comment]
You can try to work on this issue indeed. Note at this stage I suggest that you try to update the documentation in order to describe the current behavior of `Format` with too large geometry.

**Comment by olybeke** [comment]
Alright, I will keep you updated on the progress.

**Comment by Oluwabukolab** [comment]
Hi @olybeke, are you still working on this? If not, can I pick it up?🙂

**Comment by olybeke** [comment]
Hello @Bukolab99 you can go ahead with the issue.

**Comment by janithpet** [comment]
Hello, 

Would the documentation fix I made in #12477 be sufficient?

**Comment by Octachron** [comment]
The documentation and API updates in #12477 should make the behavior of `set_geometry` more regular and easier to understand.
In particular, the code snapshot in this issue will now raise an error rather than silently clip to `pp_infinity`.
Moreover, users that want to disable the `margin` and maximum indentation limit will be able to use `Format.pp_infinity` to do so.
